src/ArrayFigures.java

```diff
@@ -0,0 +1,24 @@
+public class ArrayFigures {
+
+    private Figures[] list = new Figures[10];
+    private int count;
+
+    public ArrayFigures(){
+    }
+
+    public void showfigure(){
+        for(int i = 0; i < this.count; ++i) {
+            this.list[i].getFigureName();
+        }
+    }
+
+    public void addfigure(Figures shape) {
+        if (count == list.length) {
+            System.out.println("Error!");
```

Comment:
That's not a good approach for edge case handling. If someone would use your ArrayFigures without looking into the implementation, he would never understand why I'm not allow to add another figure. In that cases you should use Exceptions and user-friendly error messages with information about what happens

Reply:
In that particular case just use collections like List to add figures and that's it